packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -5,6 +5,7 @@
 import 'dart:math' as math;
 import 'dart:ui' as ui;
 
+import 'package:flutter/rendering.dart' show FractionalOffset;
```

Comment:
This import is a layering inversion.  Instead, we should just move the definition of FractionalOffset into the painting layer.